rpc/src/optimistically_confirmed_bank_tracker.rs

```diff
@@ -194,6 +195,9 @@ impl OptimisticallyConfirmedBankTracker {
                     slot_notification_subscribers,
                     SlotNotification::OptimisticallyConfirmed(bank.slot()),
                 );
+
+                // finalize block's minimum prioritization fee cache for this bank
+                prioritization_fee_cache.finalize_priority_fee(bank.slot());
```

Comment:
> i.e. is there a chance we do not finalize priority fees because the bank is not optimistically confirmed, but is confirmed?

good point. hmm, generally speaking, there shouldn't such a chance, if we did impl this rpc subsystem right... however, iirc, we had bugs of missing optimistically-confirmed notifications for confirmed (rooted) banks.

@CriesofCarrots could you take this pr as well? as described here: https://github.com/solana-labs/solana/pull/33062#discussion_r1309490595, I think the original `.finalize_priority_fee()` is called from wrong place.

I wonder `.finalilize_priority_fee()`-ing rooted banks are worthwhile to be on the safer side.

Reply:
Ultimately `finalize_priority_fee()` should be called on rooted bank so the min priority fee available for user via rpc are accurate. Also, the timeliness of the information should also be considered. OC-ed bank could be a good compromise. 

In the unlikely event of missing OC notification for rooted bank, the result is that bank's priority fee will not be available for query, which isn't that bad from safety perspective.

Reply:
> i.e. is there a chance we do not finalize priority fees because the bank is not optimistically confirmed, but is confirmed?

I'm confused, @apfitzge . What do you mean by "confirmed" here, if not "optimistically confirmed"? Rooted?

> as described here: #33062 (comment), I think the original .finalize_priority_fee() is called from wrong place.

Oh dang, you are right. Maybe we should add some comments down in `process_notification()` so we don't make the same mistake again when trying to add other behavior that should occur on optimistic confirmation. :goldfish:

> iirc, we had bugs of missing optimistically-confirmed notifications for confirmed (rooted) banks.

We did! However, all the ones we know about are patched, right? I think it's okay for this to remain at the optimistically confirmed level.

Reply:
Sorry, I am probably using the terminology wrong, but I meant a block that's gotten 2/3+1 stake votes.
I'm not familiar with the optimistic part, so unclear if the status can skip over optimistic.

Reading more it doesn't seem like it's possible w/o a bug, so the placement seems fine to me.